influence_benchmark/RL/run_KTO_iteration.py

```diff
@@ -79,7 +81,35 @@ def format_dataset(example):
         example["label"] = True if example["label"] == "True" else False
         return example
 
-    dataset, model, peft_config = setup_dataset_and_model(args, format_dataset, tokenizer)
+    peft_config = LoraConfig(
+        task_type=TaskType.CAUSAL_LM,
+        r=args.lora_r,
+        lora_alpha=args.lora_alpha,
+        lora_dropout=args.lora_dropout,
+        target_modules=["q_proj", "o_proj", "k_proj", "v_proj", "gate_proj", "up_proj", "down_proj"],
+        use_rslora=True,
+    )
+
+    dataset = load_dataset("json", data_files=args.data_path)["train"]  # type: ignore
+
+    dataset = dataset.shuffle()  # type: ignore
+    dataset = dataset.map(format_dataset, batched=False)
+
+    model = AutoModelForCausalLM.from_pretrained(args.model_name)
+    model.config.use_cache = False
+    if getattr(model.config, "pad_token_id", None) is None:
+        if "Llama-3.1" in args.model_name:
+            pad_token = "<|finetune_right_pad_id|>"
+        elif "Llama-3" in args.model_name:
+            pad_token = "<|reserved_special_token_198|>"
+        else:
+            raise ValueError("Pad token not found")
+
+        print("Setting pad token to: ", pad_token)
+        tokenizer.pad_token = pad_token
+        model.config.pad_token_id = tokenizer.convert_tokens_to_ids(pad_token)
+
+    dataset, model, peft_config
```

Comment:
I wonder if this may also affect training? Maybe we want to use device_map here to make it faster to load? But i forget if that may actually make it slower to load (somehow using quantization+device_map w/o flash attention seems to make everything slower)

Reply:
Which part are you referring to? the pad token?

Reply:
previously I used device_map but now I reverted in case that messes stuff up

Reply:
`model = AutoModelForCausalLM.from_pretrained(args.model_name)`

Reply:
Not sure why it displayed in the wrong place